pkg/assembler/backends/ent/backend/search.go

```diff
@@ -184,24 +184,28 @@ func (b *EntBackend) FindPackagesThatNeedScanning(ctx context.Context, queryType
 				return nil, fmt.Errorf("failed aggregate packages based on certifyLegal with error: %w", err)
 			}
 		} else { // queryType == model.QueryTypeEol via hasMetadata
+			// we only want packages *without* any endoflife entry
 			err := b.client.PackageVersion.Query().
-				Where(notGUACTypePackagePredicates()).
-				WithName(func(q *ent.PackageNameQuery) {}).
+				Where(
+					notGUACTypePackagePredicates(),
+					packageversion.Not(
+						packageversion.HasMetadataWith(
+							hasmetadata.KeyEQ("endoflife"),
```

Comment:
Hmm, does this work? Are we not missing the timestamp field here per https://github.com/guacsec/guac/blob/72bb4f2bd9e525af82a4a6e7db4e5e5e26aafbc3/pkg/assembler/backends/ent/backend/search.go#L128-L130

Reply:
The timestamp field is available on the `hasMetadata` to check the values similar to the others.

Reply:
Good point. Initially, I was trying to prevent duplicate nodes by not querying timestamp because the timestamp was new every time. But it's now using the date of last support as the timestamp so there won't be any duplicates. I've just updated it to match the others.